### PR TITLE
Add .NET runtimes function starter examples

### PR DESCRIPTION
**Pull request by adityaoberai**
Add starter examples for .NET Core 3.1 and .NET 6.0 runtimes

### CODE REVIEW

**Comment by adityaoberai** [comment]
I modelled the example after the `dart-2.12` starter since the .NET SDK is not yet ready.

File: dotnet-3.1/.gitignore [hunk 1]
```diff
@@ -0,0 +1,454 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (christyjacob4) [inline dotnet-3.1/.gitignore:1]: Can we minimise this file to the bare minimum? Looks like there are too many files in here.

File: dotnet-3.1/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+### Using CLI
+
+Make sure you have [Appwrite CLI](https://appwrite.io/docs/command-line#installation) installed, and you have successfully logged into your Appwrite server. To make sure Appwrite CLI is ready, you can use the command `appwrite client --debug` and it should respond with green text `✓ Success`.
+
+Make sure you are in the same folder as your `appwrite.json` file and run `appwrite deploy function` to deploy your function. You will be prompted to select which functions you want to deploy.
+
+### Manual using tar.gz
+
+Manual deployment has no requirements and uses Appwrite Console to deploy the tag. First, enter the folder of your function. Then, create a tarball of the whole folder and gzip it. After creating `.tar.gz` file, visit Appwrite Console, click on the `Deploy Tag` button and switch to the `Manual` tab. There, set the `entrypoint` to `lib/main.dart`, and upload the file we just generated.
```
Comment (christyjacob4) [inline dotnet-3.1/README.md]: Reminder to update the entrypoint

File: dotnet-6.0/.gitignore [hunk 3]
```diff
@@ -0,0 +1,454 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (christyjacob4) [inline dotnet-6.0/.gitignore:1]: Likewise, please update this file to keep it to the bare minimum

File: dotnet-6.0/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+### Using CLI
+
+Make sure you have [Appwrite CLI](https://appwrite.io/docs/command-line#installation) installed, and you have successfully logged into your Appwrite server. To make sure Appwrite CLI is ready, you can use the command `appwrite client --debug` and it should respond with green text `✓ Success`.
+
+Make sure you are in the same folder as your `appwrite.json` file and run `appwrite deploy function` to deploy your function. You will be prompted to select which functions you want to deploy.
+
+### Manual using tar.gz
+
+Manual deployment has no requirements and uses Appwrite Console to deploy the tag. First, enter the folder of your function. Then, create a tarball of the whole folder and gzip it. After creating `.tar.gz` file, visit Appwrite Console, click on the `Deploy Tag` button and switch to the `Manual` tab. There, set the `entrypoint` to `lib/main.dart`, and upload the file we just generated.
```
Comment (christyjacob4) [inline dotnet-6.0/README.md]: Reminder to update the entrypoint

File: dotnet-3.1/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ "areDevelopersAwesome": true
+}
+```
+
+## 📝 Environment Variables
+
+List of environment variables used by this cloud function:
+
+- **APPWRITE_FUNCTION_ENDPOINT** - Endpoint of Appwrite project
+- **APPWRITE_FUNCTION_API_KEY** - Appwrite API Key
```
Comment (christyjacob4) [inline dotnet-3.1/README.md]: These environment variables are not used by this function. So you need to rewrite it to make  this section generic

File: dotnet-6.0/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ "areDevelopersAwesome": true
+}
+```
+
+## 📝 Environment Variables
+
+List of environment variables used by this cloud function:
+
+- **APPWRITE_FUNCTION_ENDPOINT** - Endpoint of Appwrite project
+- **APPWRITE_FUNCTION_API_KEY** - Appwrite API Key
```
Comment (christyjacob4) [inline dotnet-6.0/README.md]: These environment variables are not used by this function. So you need to rewrite it to make  this section generic

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by adityaoberai** [comment]
Alright 👍
I'll make the changes and push them today.

File: dotnet-3.1/.gitignore (hunk 1 above)
Comment (adityaoberai) [reply dotnet-3.1/.gitignore:1]: This is the default .gitignore file created by the .NET CLI. I can, however, try and remove some of the irrelevant files.

**Comment by adityaoberai** [comment]
@christyjacob4 can you please review the updated changes?

File: dotnet-3.1/src/Index.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  'res' variable has:
+    'Send(text, status)' - function to return text response. Status code defaults to 200
+    'Json(obj, status)' - function to return JSON response. Status code defaults to 200
+  
+  If an error is thrown, a response with code 500 will be returned.
+*/
+
+public async Task<RuntimeResponse> Main(RuntimeRequest req, RuntimeResponse res)
+{
+    // Appwrite SDK is not yet available for .NET
```
Comment (Meldiron) [inline dotnet-3.1/src/Index.cs]: Not sure if we want to mention this.

Comment (adityaoberai) [reply dotnet-3.1/src/Index.cs]: The idea was inspired from here: https://github.com/appwrite/functions-starter/blob/main/dart-2.12/lib/main.dart

I can instead mention that it is under the works and will be available soon, if that sounds alright.

**Comment by adityaoberai** [comment]
@Meldiron updated the comment.

**Review by Meldiron — APPROVED** [review]

**Pull request merged**